Propagate build_env variables from config map

chmeliik opened the pull request:
* CLOUDBLD-1228

Add variables to buildconfig, let openshift propagate them into
buildroot.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

MartinBasti commented on osbs/build/build_requestv2.py:
```diff
@@ -304,6 +304,14 @@ def set_data_from_reactor_config(self, reactor_config_data):
+        Propagates build_env variables set in config map to build environment
+        """
+        build_env = reactor_config_data.get('build_env', [])
+        self.template['spec']['strategy']['customStrategy']['env'].extend(build_env)
```
:eyes: can we trust that configmap has correct format here?

chmeliik replied:
Well it does get validated by schema, I guess I could write this in a way that is more likely to fail if we change the schema

MartinBasti replied:
It's too late for validation by atomic-rector :)

chmeliik replied:
Yeah I realized, should we do some more thorough validation here? Or is it fine the way it is now?

MartinBasti replied:
I'd say it's fine now

MartinBasti commented:
I'm thinking about blacklisting some names. like USER_PARAMS, BUILD, etc. We use for different purpose to avoid overwrites (or making sure that these won't be overwritten somehow else)

chmeliik commented:
> I'm thinking about blacklisting some names. like USER_PARAMS, BUILD, etc. We use for different purpose to avoid overwrites (or making sure that these won't be overwritten somehow else)

Couple changes:
- `set_build_env_vars()` will not set duplicate variables
- moved the method out of `set_data_from_reactor_config()`, because it is now important for it to run at the end of `render()`
- reverted old unit test and added a new one specifically for setting build_env_vars

MartinBasti commented on osbs/build/build_requestv2.py:
```diff
@@ -305,6 +310,26 @@ def set_data_from_reactor_config(self, reactor_config_data):
+                env.append({'name': var['name'], 'value': var['value']})
+                existing_vars.add(var['name'])
+                logger.info('Set environment variable from reactor config: %s', var['name'])
+            else:
```
I'd prefer fail, this warning will nobody notice and this shouldn't happen

chmeliik replied:
Good point, changed

chmeliik commented:
/packit copr-build

athos-ribeiro commented:
/packit copr-build

chmeliik commented:
Rebased, should fix py2 failure

chmeliik commented:
Release notes: Environment variables can be defined in reactor config map, they will be propagated to build environment
